src/TpetraLinearSystem.C

```diff
@@ -1649,12 +1649,12 @@ bool TpetraLinearSystem::checkForZeroRow(bool useOwned, bool doThrow, bool doPri
     if (global_row_exists[ii] && bulkData.parallel_rank() == 0 && row_sum < 1.e-10) {
       found = true;
       GlobalOrdinal gid = ii+1;
-      stk::mesh::EntityId nid = GLOBAL_ENTITY_ID(gid, numDof_);
+      stk::mesh::EntityId nid = (gid - 1) / numDof_ + 1;
       stk::mesh::Entity node = bulkData.get_entity(stk::topology::NODE_RANK, nid);
```

Comment:
Hmm, I think your change is ok, but I didn't realize we had code like this. That's a weird way of coming up with a global node id.

Reply:
@alanw0 I'm happy to confess that I do not know why this is how the code computes the entityId, but I hope that this is compatible with stk's algorithms!